### ISSUE TITLE
'nan' values in emission lines in Orion and fibers with inf values

### ISSUE THREAD

**Issue by JEMendezD** [sdss/lvmdrp#112]
The following observations associated with the Orion Nebula:

Tile ID  |  MJD  |  EXP
--- | --- | ---
1028655  |  60285  |  8910
1028655  |  60296  |  9478
-------  |  60277  |  8487

Present the following issues with the new DRP reduction:

1. Many of the fibers show inf values. Although I understand this may be an effect of dead fibers, in the previous version of the DRP, the fibers with inf values were fine. They seem associated with bright knots covering O stars.

2. Many of the lines (e.g., HI 4861) present pixels with 'nan' values in fibers that are not dead. 

<img width="738" alt="Screenshot 2024-06-19 at 18 59 30" src="https://github.com/sdss/lvmdrp/assets/17130010/9df59952-5453-4e28-a15a-a307b48dc74f">

**Comment by JEMendezD** [comment]
It seems that this problem is not only present in Orion but in some bright regions of M8 and M20. It could be a general problem of the brightest regions.

**Comment by ndrory** [comment]
We should check the mask extension (best to look in the original 2D mask) to see if these pixels were masked by the cosmic ray rejection routine. I tested it on arc lines to make sure it does not identify bright lines, but these ones might be even brighter than what I tested on.

**Comment by JEMendezD** [comment]
Yes, these pixels match with those of the mask extension. Perhaps this problem can be solved by using a higher cosmic ray rejection threshold?

**Comment by ndrory** [comment]
And we’re sure it’s from the cosmic ray rejection? We should make sure by checking the 2D masks before and after cosmics.

N.


— Niv Drory —
McDonald Observatory, The University of Texas at Austin
https://www.as.utexas.edu/~drory

On Jun 20, 2024, at 16:44, José Eduardo Méndez-Delgado ***@***.***> wrote:

﻿

Yes, these pixels match with those of the mask extension. Perhaps this problem can be solved by using a higher cosmic ray rejection threshold?

**Comment by JEMendezD** [comment]
I have reduced the pointing with the key '-d': drp run -m 60285 -de 8910.

Upon reviewing the lvm-dobject* files present in the 'ancillary' folder, I checked the masks for bad and saturated pixels, and they do not seem to match the NaNs and Infs I observe in the SFrame.

**Comment by tonywong94** [comment]
This could be related to my request in #95 that bad CCD columns be blanked in the FLUX image.  Of course, I now realize that these are set to 1 in the MASK image, so if people would prefer that the FLUX image not have NaNs, we might want to roll that change back.  I can just make sure to apply the mask before collapsing the spectra.

**Comment by JEMendezD** [comment]
Hello,

Following Niv and Kathryn's suggestion, I have been reviewing the ImageMethod routine in the DRP. I changed the threshold value from 1.3 to 2.0. After the local reduction of Orion, the "NaNs" that appeared in the emission lines practically disappeared, except for a few spots in H-alpha, the brightest line of all.


<img width="1502" alt="Screenshot 2024-07-03 at 08 43 45" src="https://github.com/sdss/lvmdrp/assets/17130010/8b2c1b20-339b-404f-8c80-209b74770959">


Visually, I inspected the cosmic rays, and it seems that their distribution remains the same, meaning that the increase from 1.3 to 2.0 does not appear to have caused the cosmic rays to go unnoticed. I will be testing if there are rlim values that can completely eliminate the problematic NaNs in the emission lines without causing defects in the cosmic ray masking and will report here if I find any.

On the other hand, the inf values in some fibers remain unchanged. This seems to be related to a different issue not related to the cosmic ray routine.
